### PR TITLE
Create reusable component to demo how to apply Tip to truncated Text in DataTable

### LINKED ISSUE #1503: Add examples of tooltip within a Table for truncated text
Table research/guidance on truncated text 
https://www.figma.com/file/XEoIWIlLfcARRIFhkpFCch/HPE-Table?node-id=2390%3A12752

https://design-system.hpe.design/components/table
image

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
Creates `TruncatedText` reusable component that will apply a tooltip to text when it has been truncated. Applies to appropriate DataTable examples.

#### Where should the reviewer start?
aries-site/src/examples/components/table/TruncatedText.js

#### What testing has been done on this PR?
Tested local in DS site.

#### How should this be manually tested?
In deploy, hover over a cell with "...", should see tooltip. For cell that isn't truncated, no tooltip should appear.

#### What are the relevant issues?
Closes #1503 

#### Should this PR be mentioned in Design System updates?
No.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

File: aries-site/src/examples/components/table/TruncatedText.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Text } from 'grommet';
+
+export const TruncatedText = ({ value }) => {
+  const ref = useRef();
+  const [tip, setTip] = useState();
+
+  useEffect(() => {
+    if (ref.current && ref.current.scrollWidth > ref.current.offsetWidth)
```
Comment (MikeKingdom) [inline aries-site/src/examples/components/table/TruncatedText.js:10]: Should this also drop the tip if not needed at some point?

Comment (taysea) [reply aries-site/src/examples/components/table/TruncatedText.js:10]: Good thought. I'll update the condition

**Comment by ericsoderberghp** [comment]
I'm wondering if we should build this directly into the Text component, so callers don't have to do something like TruncatedText. Perhaps: `<Text truncate="tip" />` or `<Text truncate tip />`?

**Comment by taysea** [comment]
> I'm wondering if we should build this directly into the Text component, so callers don't have to do something like TruncatedText. Perhaps: `<Text truncate="tip" />` or `<Text truncate tip />`?

truncate="tip" was exactly what I was going to propose as a grommet enhancement after putting this together yesterday : ) wasn't sure if we wanted to merge this and then enhance grommet or just enhance grommet right away. preference? @ericsoderberghp

**Comment by ericsoderberghp** [comment]
> > I'm wondering if we should build this directly into the Text component, so callers don't have to do something like TruncatedText. Perhaps: `<Text truncate="tip" />` or `<Text truncate tip />`?
> 
> truncate="tip" was exactly what I was going to propose as a grommet enhancement after putting this together yesterday : ) wasn't sure if we wanted to merge this and then enhance grommet or just enhance grommet right away. preference? @ericsoderberghp

your choice :)

**Comment by taysea** [comment]
> your choice :)

Filed a PR: https://github.com/grommet/grommet/pull/5453

I'm out next week, so up to you and @MikeKingdom which direction to move forward with : )

**Comment by MikeKingdom** [comment]
> > your choice :)
> 
> Filed a PR: [grommet/grommet#5453](https://github.com/grommet/grommet/pull/5453)
> 
> I'm out next week, so up to you and @MikeKingdom which direction to move forward with : )

I see how it goes :) No problem.

**Pull request merged**